core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1831,6 +1834,11 @@ private void commitIndexWriter(final IndexWriter writer, final Translog translog
         }
     }
 
+    // commit hook for testing
+    void callCommitOnWriter(IndexWriter writer) throws IOException {
```

Comment:
I think we can achieve the same without introducing this method (that has potential to be used in the wrong place) with the following:

```diff
diff --git a/core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java b/core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java
index 18fafb6e90..8bbf320e26 100644
--- a/core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java
+++ b/core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java
@@ -1778,7 +1778,7 @@ public class InternalEngine extends Engine {
      * @param syncId   the sync flush ID ({@code null} if not committing a synced flush)
      * @throws IOException if an I/O exception occurs committing the specfied writer
      */
-    private void commitIndexWriter(final IndexWriter writer, final Translog translog, @Nullable final String syncId) throws IOException {
+    void commitIndexWriter(final IndexWriter writer, final Translog translog, @Nullable final String syncId) throws IOException {
         ensureCanFlush();
         try {
             final long localCheckpoint = seqNoService().getLocalCheckpoint();
@@ -1810,7 +1810,7 @@ public class InternalEngine extends Engine {
                 return commitData.entrySet().iterator();
             });
 
-            callCommitOnWriter(writer);
+            writer.commit();
         } catch (final Exception ex) {
             try {
                 failEngine("lucene commit failed", ex);
@@ -1837,11 +1837,6 @@ public class InternalEngine extends Engine {
         }
     }
 
-    // commit hook for testing
-    void callCommitOnWriter(IndexWriter writer) throws IOException {
-        writer.commit();
-    }
-
     private void ensureCanFlush() {
         // translog recover happens after the engine is fully constructed
         // if we are in this stage we have to prevent flushes from this
diff --git a/core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java b/core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
index a767e50d4e..a10381be4a 100644
--- a/core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
+++ b/core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
@@ -1163,8 +1163,9 @@ public class InternalEngineTests extends ESTestCase {
     }
 
     public void testSyncedFlush() throws IOException {
-        try (Store store = createStore();
-             Engine engine = new InternalEngine(config(defaultSettings, store, createTempDir(), new LogByteSizeMergePolicy(), null))) {
+        try (
+                Store store = createStore();
+                Engine engine = new InternalEngine(config(defaultSettings, store, createTempDir(), new LogByteSizeMergePolicy(), null))) {
             final String syncId = randomUnicodeOfCodepointLengthBetween(10, 20);
             ParsedDocument doc = testParsedDocument("1", null, testDocumentWithTextField(), B_1, null);
             engine.index(indexForDoc(doc));
@@ -2496,8 +2497,8 @@ public class InternalEngineTests extends ESTestCase {
             final Path translogPath = createTempDir();
             try (InternalEngine engine = new InternalEngine(config(defaultSettings, store, translogPath, newMergePolicy(), null, null)) {
                 @Override
-                void callCommitOnWriter(IndexWriter writer) throws IOException {
-                    super.callCommitOnWriter(writer);
+                void commitIndexWriter(IndexWriter writer, Translog translog, String syncId) throws IOException {
+                    super.commitIndexWriter(writer, translog, syncId);
                     if (throwErrorOnCommit.get()) {
                         throw new RuntimeException("power's out");
                     }
```

What do you think?